text_generation/causal_lm/cpp/generate_pipeline/generation_config.hpp

```diff
@@ -0,0 +1,219 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstdlib>
+#include <functional>
+#include <nlohmann/json.hpp>
+#include <fstream>
+#include <group_beam_searcher.hpp>  // used only for StopCriteria
+#include <limits>
+
+// forward declaration
+class Sequence;
+
+// forward declaration
+class LLMPipeline;
+
+// Similar to HuggingFace GenerationConfig
+struct GenerationConfig {
+    // todo: add copy constructor
+    
+    // Generic
+    size_t m_max_new_tokens = SIZE_MAX;
+    size_t m_max_length = SIZE_MAX; // m_max_new_tokens should have priority over m_max_length
+    bool m_ignore_eos = false;
+    int64_t m_eos_token = 2; // There's no way to extract special token values from the tokenizer for now
```

Comment:
It is stored in (de)tokenizer `rt_info` under `eos_token_id` key.

Reply:
we need it in both places. E.g. beam search should know when to stop, and beam_search can be called with raw input_ids, and config without any information about tokenization.

At the same time we should keep this information in tokenizer as well, since it might be needed by streamer.